### PR TITLE
Assignment 1 - Interactive shell

### PR DESCRIPTION
**Pull request by apapapap**
Added code to run the program as an interactive shell, the shell supports all the commands whose executable file(s) are found in the `$PATH` environment variable.

Below is the sample output of the program:
```
ashish@velotio-ThinkPad-E14-Gen-2:~/work/Projects/velotio/go-k8s-training$ go run assignment-1-shell/main.go 
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish/work/Projects/velotio/go-k8s-training$ ls
assignment-1-shell  README.md
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish/work/Projects/velotio/go-k8s-training$ 
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish/work/Projects/velotio/go-k8s-training$ 
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish/work/Projects/velotio/go-k8s-training$ history
1  ls
2  history
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish/work/Projects/velotio/go-k8s-training$ ls -la
total 44
drwxrwxr-x 5 ashish ashish 4096 Jun 15 10:23 .
drwxrwxr-x 3 ashish ashish 4096 Jun 15 10:01 ..
drwxrwxr-x 2 ashish ashish 4096 Jun 15 10:03 assignment-1-shell
drwxrwxr-x 8 ashish ashish 4096 Jun 15 10:50 .git
-rw-rw-r-- 1 ashish ashish  269 Jun 15 10:01 .gitignore
drwxrwxr-x 3 ashish ashish 4096 Jun 15 10:23 .history
-rw-rw-r-- 1 ashish ashish   34 Jun 15 10:01 README.md
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish/work/Projects/velotio/go-k8s-training$ invalidCmd
Error running command: invalidCmd
exec: "invalidCmd": executable file not found in $PATH
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish/work/Projects/velotio/go-k8s-training$ history
1  ls
2  history
3  ls -la
4  invalidCmd
5  history
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish/work/Projects/velotio/go-k8s-training$ cd ..
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish/work/Projects/velotio$ cd go-k8s-training 
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish/work/Projects/velotio/go-k8s-training$ cd invalidDir
Error running command: [cd invalidDir]
chdir invalidDir: no such file or directory
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish/work/Projects/velotio/go-k8s-training$ cd 
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish$ 
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish$ history
1  ls
2  history
3  ls -la
4  invalidCmd
5  history
6  cd ..
7  cd go-k8s-training
8  cd invalidDir
9  cd
10 history
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish$ 
ashish@velotio-ThinkPad-E14-Gen-2:~/home/ashish/work/Projects/velotio/go-k8s-training$ exit

ashish@velotio-ThinkPad-E14-Gen-2:~/work/Projects/velotio/go-k8s-training$ 
```

### CODE REVIEW

File: assignment-1-shell/main.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+func changeDir(cmdStrArr []string) {
+	newDir := ""
+	var err error
+	if len(cmdStrArr) > 1 {
+		newDir = cmdStrArr[1]
+	} else {
+		newDir, err = os.UserHomeDir()
+		if err != nil {
+			errMsg := fmt.Sprintf("Error running command: %v\n%v", cmdStrArr, err)
+			fmt.Fprintln(os.Stderr, errMsg)
```
Comment (abhishekdwivedi3060) [inline assignment-1-shell/main.go:102]: nit: can return from here

File: assignment-1-shell/main.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		if err != nil {
+			errMsg := fmt.Sprintf("Error running command: %s\n%v", cmdStr, err)
+			fmt.Fprintln(os.Stderr, errMsg)
+		}
+	}
+}
+
+func printHistory() {
+	for _, cmd := range cmdHistory {
+		fmt.Printf("%v", cmd)
```
Comment (abhishekdwivedi3060) [inline assignment-1-shell/main.go]:
```suggestion
		fmt.Println("%v", cmd)
```
Would be better to print the sequence number as well

**Review by abhishekdwivedi3060 — CHANGES_REQUESTED** [review]

File: assignment-1-shell/main.go (hunk 2 above)
Comment (apapapap) [reply assignment-1-shell/main.go]: Done

File: assignment-1-shell/main.go (hunk 1 above)
Comment (apapapap) [reply assignment-1-shell/main.go:102]: Done

**Pull request closed without merging**